### PR TITLE
Update UpsertColumn to include an error response

### PR DESCRIPTION
**Pull request by Tang8330**
Adding this in so we can have an extra guardrail to address https://github.com/artie-labs/transfer/pull/955#discussion_r1802191197

### CODE REVIEW

File: clients/shared/merge.go [hunk 1]
```diff
@@ -90,9 +90,14 @@ func Merge(ctx context.Context, dwh destination.DataWarehouse, tableData *optimi
 		for attempts := 0; attempts < backfillMaxRetries; attempts++ {
 			backfillErr = BackfillColumn(dwh, col, tableID)
 			if backfillErr == nil {
-				tableConfig.Columns().UpsertColumn(col.Name(), columns.UpsertColumnArg{
+				err = tableConfig.Columns().UpsertColumn(col.Name(), columns.UpsertColumnArg{
 					Backfilled: typing.ToPtr(true),
 				})
+
+				if err != nil {
```
Comment (danafallon) [inline clients/shared/merge.go:97]: Could inline this

File: lib/typing/columns/columns_test.go [hunk 2]
```diff
@@ -194,33 +194,33 @@ func TestColumns_UpsertColumns(t *testing.T) {
...
 		PrimaryKey: typing.ToPtr(true),
-	})
+	}))
 	aaaCol, _ := cols.GetColumn("aaa")
 	assert.True(t, aaaCol.ToastColumn)
 	assert.True(t, aaaCol.primaryKey)
 	assert.Equal(t, aaaCol.KindDetails, typing.Invalid)
 
 	length := len(cols.columns)
 	for i := 0; i < 500; i++ {
```
Comment (danafallon) [inline lib/typing/columns/columns_test.go:222]: How come we're doing this 500 times?

File: clients/shared/merge.go (hunk 1 above)
Comment (danafallon) [reply clients/shared/merge.go:97]: Although with the multiple lines that could make it hard to read, so I'm also good with leaving it as is.

**Review by danafallon — APPROVED** [review]

File: lib/typing/columns/columns_test.go (hunk 2 above)
Comment (Tang8330) [reply lib/typing/columns/columns_test.go:222]: Lol, I was testing it to be idempotent. It's not needed though, I'll remove.

**Pull request merged**